context/app/static/js/components/entity-search/MetadataMenu/MetadataMenu.jsx

```diff
@@ -37,11 +38,38 @@ function NotebookMenuItem(props) {
   );
 }
 
-function MetadataMenu({ entityType }) {
+const errorHelper = {
+  datasets: (rowCount) =>
+    `You have selected ${rowCount} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`,
+  protectedDataset: (rows) =>
+    `You have selected a protected dataset (${rows[0]._source.hubmap_id}). Workspaces currently only supports published public datasets. To remove the protected dataset from workspace creation, click the “Remove Protected Datasets” button below or return to the previous screen to manually remove this dataset.`,
+  protectedDatasets: (rows) =>
+    `You have selected ${rows.length} protected datasets. Workspaces currently only supports published public datasets. To remove protected datasets from this workspace creation, click the “Remove Protected Datasets” button below or return to the previous screen to manually remove those datasets.`,
```

Comment:
Should we just pass rows for consistency? Both `datasets` and `protectedDatasets` depend on the length of the rows array.

Reply:
I decided to pass the size here initially because `selectedRows` is a set, while `protectedRows` is an array; I didn't like `rows.size` in one case and `rows.length` in the other.

I've applied this revision but modified the arg names and added an explanatory comment to ensure clarity re: why this difference is here.